src/Collector/Finder/VariableFinder.php

```diff
@@ -44,11 +54,10 @@ public function __construct(CollectedDataNode $collectedDataNode, TypeSerializer
      */
     public function find(string $className, string $methodName): array
     {
-        return array_merge(
-            $this->collectedVariables[$className][''] ?? [],
-            $this->findInParents($className),
+        return Variable::merge(
```

Comment:
This is confusing for me. Why Variable has static merge method where all other variables are somehow merged? I would create some service with non static method or at least helper.